### PR TITLE
move types from wsr

### PR DESCRIPTION
**Pull request by mosheKerbel**
### 🔦 Summary

...

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [ ] Visual test

### CODE REVIEW

**Comment by simast** [comment]
:clap:
Could you also update our generator codemod? As I noticed that with this PR we will start exposing all exported component types from `src/index.d.ts`:

https://github.com/wix/wix-style-react/blob/9220a0d5f0cc4bb1aa8d77311fd0c95480f16068/.wuf/generator/codemods/index-ts-file.js#L9

**Comment by mosheKerbel** [comment]
> 👏
> Could you also update our generator codemod? As I noticed that with this PR we will start exposing all exported component types from `src/index.d.ts`:
> 
> https://github.com/wix/wix-style-react/blob/9220a0d5f0cc4bb1aa8d77311fd0c95480f16068/.wuf/generator/codemods/index-ts-file.js#L9

thanks for the heads up 👍 
will do it.

**Comment by mosheKerbel** [comment]
@giladsegal 
List of already existing types but without drivers types:
- [ ] Card/Subhader.uni.driver
- [ ] Card/Header.uni.driver
- [ ] Badge.uni.driver
- [ ] Divider.uni.driver
- [ ] Palette.uni.driver
- [ ] BarChart.uni.driver
- [ ] Calendar.uni.driver
- [ ] Calendar.driver
- [ ] DatePicker.driver
- [ ] BadgeSelect.uni.driver
- [ ] Breadcrumbs.driver
- [ ] Brreadcrumbs.uni.driver
- [ ] CounterBadger.driver
- [ ] MediaOverlay.uni.drover
- [ ] PreviewWidget.uni.driver
- [ ] MarketingLayout.uni.driver
- [ ] MobilePreviewWidget.uni.driver

File: src/Badge/Badge.uni.driver.d.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { BaseUniDriver } from 'wix-ui-test-utils/unidriver';
+
+export interface BadgeUniDriver extends BaseUniDriver {
+  getContent: () => Promise<string | null>,
+  text: () => Promise<string>;
+  getType: () => Promise<string>;
```
Comment (giladsegal) [inline src/Badge/Badge.uni.driver.d.ts]: I would use `BadgeType`, `BadgeSkin`, and `BadgeSize` as the value or the resolved promises

File: src/Badge/Badge.uni.driver.d.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  getSkin: () => Promise<string>;
+  getSize: () => Promise<string>;
+  isUppercase: Promise<boolean>;
+  hasClickCursor: Promise<boolean>;
```
Comment (giladsegal) [inline src/Badge/Badge.uni.driver.d.ts]: should be a function

File: src/Badge/Badge.uni.driver.d.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { BaseUniDriver } from 'wix-ui-test-utils/unidriver';
+
+export interface BadgeUniDriver extends BaseUniDriver {
+  getContent: () => Promise<string | null>,
+  text: () => Promise<string>;
+  getType: () => Promise<string>;
+  getSkin: () => Promise<string>;
+  getSize: () => Promise<string>;
+  isUppercase: Promise<boolean>;
```
Comment (giladsegal) [inline src/Badge/Badge.uni.driver.d.ts]: should be a function

File: src/Card/Subheader/Subheader.uni.driver.d.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { BaseUniDriver } from 'wix-ui-test-utils/unidriver';
+
+export interface SubheaderUniDriver extends BaseUniDriver {
+  title(): Promise<string>;
+  titleNode(): Promise<HTMLElement>;
```
Comment (giladsegal) [inline src/Card/Subheader/Subheader.uni.driver.d.ts]: should be `titleNode(): UniDriver`
same applies for `suffixNode`

File: src/Calendar/Calendar.uni.driver.d.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import { BaseUniDriver } from 'wix-ui-test-utils/unidriver';
+
+export interface CalendarUniDriver extends BaseUniDriver {
+  exists: () => Promise<boolean>;
+  close: () => Promise<any>;
+  isVisible: () => Promise<boolean>;
+  getCurrentMonthWithYear: () => Promise<string>;
+  getMonthAndYear:() => Promise<Array<string>>;
+  getNthWeekDayName: (n:number) => Promise<HTMLElement>;
```
Comment (giladsegal) [inline src/Calendar/Calendar.uni.driver.d.ts]: return value is UniDriver not Promise<HTMLelement>

File: src/Calendar/Calendar.uni.driver.d.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  isHeaderVisible: () => Promise<boolean>;
+  isYearDropdownExists: () => Promise<boolean>;
+  isYearCaptionExists: () => Promise<boolean>;
+  isMonthDropdownExists: () => Promise<boolean>;
+  isMonthCaptionExists: () => Promise<boolean>;
+  getMonthCaption: () => Promise<string>;
+  getMonthDropdownLabel: () => Promise<string>;
+  getSelectedYear: () => Promise<string>;
+  getFocusedDay: () => Promise<string | null>;
+  getFocusedDayElement: () => Promise<HTMLElement>;
```
Comment (giladsegal) [inline src/Calendar/Calendar.uni.driver.d.ts]: same here

File: src/MediaOverlay/MediaOverlay.uni.driver.d.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import { BaseUniDriver } from 'wix-ui-test-utils/unidriver';
+
+export interface MediaOverlayUniDriver extends BaseUniDriver {
+  hover: () => Promise<void>;
+  getMediaUrl: () => Promise<string | null>;
+  getMediaNode: () => Promise<HTMLElement | null>;
```
Comment (giladsegal) [inline src/MediaOverlay/MediaOverlay.uni.driver.d.ts]: Media layout interface should be generic of type T
`getNative` used in `getMediaNode` implementation will return `HTMLElement` for jsDOM drivers, and will return `ReactWrapper` for enzyme drivers

File: src/DataTable/index.d.ts [hunk 8]
```diff
@@ -81,6 +84,9 @@ export type DataTableColumn = {
   sortable?: boolean;
   sortDescending?: boolean;
   style?: React.CSSProperties;
-  infoTooltipProps?: any; // TODO: replace with OmitPolyfill<TooltipProps, 'dataHook' | 'moveBy'>
+  infoTooltipProps?: OmitPolyfill<
```
Comment (mosheKerbel) [inline src/DataTable/index.d.ts]: remove upgrade

Comment (mosheKerbel) [inline src/DataTable/index.d.ts]: remove upgrade

File: src/ImageViewer/index.d.ts [hunk 9]
```diff
@@ -8,8 +10,8 @@ export interface ImageViewerProps {
    * @deprecated
    * @see tooltipProps
    */
-  tooltipPlacement?: any; // TODO: replace with TooltipProps["placement"];
-  tooltipProps?: any; // TODO: replace with OmitPolyfill<TooltipProps, "content">;
+  tooltipPlacement?: TooltipNewProps['placement'];
+  tooltipProps?: OmitPolyfill<TooltipNewProps, 'content' | 'upgrade'>;
```
Comment (mosheKerbel) [inline src/ImageViewer/index.d.ts]: dataHook

File: testkit/enzyme.d.ts [hunk 10]
```diff
@@ -30,6 +30,7 @@ import {CarouselDriver} from '../src/Carousel/Carousel.driver';
 import {CheckboxDriver} from '../src/Checkbox/Checkbox.driver';
 import {CircularProgressBarDriver} from '../src/CircularProgressBar/CircularProgressBar.driver';
 import {CloseButtonUniDriver} from '../src/CloseButton/CloseButton.uni.driver';
+import {ColorInputUniDriver} from '../src/ColorInput/ColorInput.uni.driver';
```
Comment (mosheKerbel) [inline testkit/enzyme.d.ts]: need to remove the auto generated comment on top of the 3 testkit files

**Comment by shlomitc** [comment]
👏

**Pull request merged**